pkg/querier/worker/worker.go

```diff
@@ -92,9 +101,19 @@ type querierWorker struct {
 	managers map[string]*processorManager
 
 	metrics *Metrics
+
+	grpcClientConfig      grpcclient.Config
+	maxConcurrentRequests int
 }
 
 func NewQuerierWorker(cfg Config, rng ring.ReadRing, handler RequestHandler, logger log.Logger, reg prometheus.Registerer, codec RequestCodec) (services.Service, error) {
+	if !cfg.UseSeparatedGRPCClients {
+		level.Warn(logger).Log("msg", "Using the same client for frontend and scheduler. This is deprecated and will be removed in the future. Please use separated clients for frontend and scheduler.")
```

Comment:
IF this is deprecated, we should add a note to the docs.

Reply:
Are we sure we want to deprecate this? IIUC 99.% of the users will only need one grpc client. So we will require them to configure two identical grpc configs one for the frontend and another for the scheduler?

Reply:
great point. https://github.com/grafana/loki/pull/12726/commits/35820954f379ea91a62b369a88615125b9d20ed9 reverts it.